coderd/database/dbauthz/dbauthz.go

```diff
@@ -4218,6 +4226,10 @@ func (q *querier) GetAuthorizedWorkspaces(ctx context.Context, arg database.GetW
 	return q.GetWorkspaces(ctx, arg)
 }
 
+func (q *querier) GetAuthorizedWorkspacesAndAgentsByOwnerID(ctx context.Context, ownerID uuid.UUID, _ rbac.PreparedAuthorized) ([]database.GetWorkspacesAndAgentsByOwnerIDRow, error) {
```

Comment:
Why are we dropping the prepared auth here?  It calls into the normal function which re-prepares the auth.

Reply:
My understanding is that we need an implementation of `GetAuthorized...` on `dbauthz` to fulfill the `database.Store` interface. There's a TODO to delete these Authorized functions, but it's from a few years ago.

```
func (q *querier) GetAuthorizedWorkspaces(ctx context.Context, arg database.GetWorkspacesParams, _ rbac.PreparedAuthorized) ([]database.GetWorkspacesRow, error) {
	// TODO Delete this function, all GetWorkspaces should be authorized. For now just call GetWorkspaces on the authz querier.
	return q.GetWorkspaces(ctx, arg)
}
```

However this does make me think we shouldn't be using the `GetAuthorized...` variant at all - we should just be calling the regular, and letting that construct the prep. 

`/workspaces` currently calls `GetAuthorizedWorkspaces`, but switching it to call `GetWorkspaces` doesn't appear to break anything as the prep gets constructed in `dbauthz` anyway, so we can fix that too.

Reply:
Yeah, I think it was for back-compatibility that we added the `GetAuthorized...` variants.